src/components/homepage/Homepage.jsx

```diff
@@ -1,11 +1,20 @@
+import React from 'react';
 import Movie from "../movie/Movie";
 import { MovieContainer } from "./Styles/StyleMovieContainer";
 import { GetMovies } from "../../API/movies";
 import { useEffect, useState } from "react";
+import { GetMoviesBasedOnId } from "../../API/movies";
+
+export const MovieContext = React.createContext();
```

Comment:
Can you refactor the context creation code to a separate file? Only use the context in a component where it is needed.

Reply:
Done